packages/central-server/src/hooks/utilities.js

```diff
@@ -34,3 +34,17 @@ export function parseCoordinates(answerText) {
 
   return { latitude, longitude };
 }
+
+export function updateJsonFieldValue(currentJsonValues, newKeyValues) {
+  if (Object.keys(currentJsonValues).length > 0) {
+    const updatedKeyValues = { ...currentJsonValues };
+    const newKeys = Object.keys(newKeyValues);
+
+    newKeys.forEach(key => {
+      updatedKeyValues[key] = newKeyValues[key];
+    });
+
+    return updatedKeyValues;
+  }
+  return newKeyValues;
+}
```

Comment:
Nice solve, but it's actually even easier:
```suggestion

export function updateJsonFieldValue(currentJsonValues, newKeyValues) {
 return { ...currentJsonValues, ...newKeyValues };
}
```
Magic huh!? 
In fact, this is so straightforward I'd suggest not even making a util function here, and just doing the above line wherever it's needed :-)

Reply:
Note that this will work even if currentJsonValues is undefined ✨